tests/func/test_gc.py

```diff
@@ -204,3 +204,12 @@ def test_gc_no_dir_cache(tmp_dir, dvc, repo_template):
 
 def _count_files(path):
     return sum(len(files) for _, _, files in os.walk(path))
+
+def test_gc_no_unpacked_dir(tmp_dir, dvc, repo_template):
+    dir_stage, = dvc.add(["foo"])
+
+    assert _count_files(dvc.cache.local.cache_dir) == 1
+    os.unlink(dir_stage.outs[0].cache_path)
```

Comment:
You are removing the cache file, but the dvc file stays, so gc is going to count it as used. Please see the reproducer script in the issue, it works a bit differently, right?

Reply:
Ok. I have updated the test. Now the test replicates the reproducer script in the issue.